### DISCUSSION TITLE
When importing  csv file keeps giving me an E_ERROR

### DISCUSSION THREAD

**Discussion by Mimi154** [code4recovery/12-step-meeting-list#1443] [Q&A]
Help! I am trying to update our local meetings. I want to delete all existing meetings and upload the new CSV file. I have attached the file, a screenshot of the error, and the error details.

![WP Backend  Error](https://github.com/code4recovery/12-step-meeting-list/assets/172465777/c9b173bf-fd92-4ad6-ba5f-a1beb8e99e3b)

[template New Meeting List 5_30_24.csv](https://github.com/user-attachments/files/15798479/template.New.Meeting.List.5_30_24.csv)

Upload CSV When importing...
_don't delete anything
_delete only the meetings, locations, and groups for the regions present in this CSV
_delete all meetings, locations, groups, districts, and regions

I have tried all three options and still receive the same error. Any feedback would be greatly appreciated.

Thanks!

WordPress version 6.5.3
Active theme: Spacious (version 1.9.6)
Current plugin: 12 Step Meeting List (version 3.14.35)
PHP version 7.4.33

Error Details
=============
An error of type E_ERROR was caused in line 1475 of the file /var/www/ca4la/wp-content/plugins/12-step-meeting-list/includes/functions.php. Error message: Uncaught Error: Call to undefined function mb_detect_encoding() in /var/www/ca4la/wp-content/plugins/12-step-meeting-list/includes/functions.php:1475
Stack trace:
#0 [internal function]: {closure}()
#1 /var/www/ca4la/wp-content/plugins/12-step-meeting-list/includes/functions.php(1478): array_walk_recursive()
#2 /var/www/ca4la/wp-content/plugins/12-step-meeting-list/includes/admin_import.php(98): tsml_import_buffer_set()
#3 /var/www/ca4la/wp-includes/class-wp-hook.php(324): tsml_import_page()
#4 /var/www/ca4la/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()
#5 /var/www/ca4la/wp-includes/plugin.php(517): WP_Hook->do_action()
#6 /var/www/ca4la/wp-admin/admin.php(259): do_action()
#7 /var/www/ca4la/wp-admin/edit.php(10): require_once('/var/www/ca4la/...')
#8 {main}
  thrown

**Comment by SIA-WEBSUP** [comment]
Would you provide the URL to your site so we can pull some diagnostic info from it?

**Reply by SIA-WEBSUP** [reply]
> Would you provide the URL to your site so we can pull some diagnostic info from it?

Is this is https://ca4la.org/meetings/ ?

**Reply by Mimi154** [reply]
Yes, that is the site.

**Comment by kiyote33** [comment]
Hi @Mimi154, we've been unable to reproduce the error you are getting in a test environment mimicking the environment you're reporting to use (WordPress version 6.5.3, Active theme: Spacious (version 1.9.6), Current plugin: 12 Step Meeting List (version 3.14.35), and PHP (version 7.4.33)). That makes me think that one of the plugins in your environment might be affecting the import. Could you try deactivating the other plugins (one at a time) to see which one might be the culprit. Of course, you will need to run the import after each deactivation to see if the meeting import works. If you have no meetings already loaded, then use the **don't delete anything** option, otherwise simply use the **delete only the meetings, locations, and groups for the regions present in this CSV** option. Let us know what you find.

**Comment by SIA-WEBSUP** [comment]
@Mimi154, 

The most recent release of the plugin (3.14.36) has a workaround to the issue you were running into. 

If you would, please update to this version of the plugin and let us know if it resolves the issue for you.
